### PR TITLE
Unify YAML indent

### LINKED ISSUE #1546: GitHub Actions docs: inconsistency of YAML indentation between dashes
See this question:
https://stackoverflow.com/questions/17014460/yaml-indentation-for-array-in-hash

The docs are not consistent regarding YAML indentation between dashes.

Example:
[Using Flake8 to lint code](https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#using-flake8-to-lint-code):
```yaml
steps:
- uses: actions/checkout@v2
- name: Set up Python
  uses: actions/setup-python@v2
  with:
    python-version: '3.x'
...
```

[Running tests with tox](https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#running-tests-with-tox):
```yaml
    steps:
      - uses: actions/checkout@v2
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: ${{ matrix.python }}
...
```

There are many other examples of inconsistency, between different snippets and sometimes within a single snippet. That's confusing, I suggest to agree on a convention and stick to it.

### PR DESCRIPTION
**Pull request by Simran-B**
### Why:

Closes #1546

### What's being changed:

- Indent nested lists with two spaces before the dash to comply to [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md#indentation) and [YAML standard](https://yaml.org/spec/1.2/spec.html#id2760167)
  ```diff
       steps:
  -    - run: ...
  +      - run: ...
  ```
- Indent blocks with no more than two spaces
  ```diff
       steps:
         - run: |
  -            command
  +          command
  ```
- Fix wrong indentation in one of the examples
  ```diff
       steps:
         - uses: ...
  -      with: ...
  +        with: ...
  ```
- Remove some unnecessary whitespace
- Add `steps` key to one of the examples for context
- Change indentation from 3 to 2 spaces in a YAML frontmatter
- Fix a typo

Regular expressions I used to discover indentation issues (with a few false-positives):
- `^( +)[a-z].*:(\{% raw %\})?\r?\n\1(-|#.+\r?\n\1-)` (find unintended nested lists)
- `^( +)[a-z].+: *\|\s*\r?\n\1[ ]{3,}` (find too much indented block content)
- `^( +)- [a-z].*:(\{%.+%\})?\r?\n\1[a-z]` (find mappings with broken indentation)

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (eg. a "before and after video")

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks so much @Simran-B! I'll get this triaged for review 🎉

**Review by skedwards88 — DISMISSED** [review]
✨ Thank you for cleaning up the docs! I'll get this merged down for you.

**Review by skedwards88 — APPROVED** [review]
re-approving after resolving merge conflict

**Pull request merged**
